compiler/rustc_typeck/src/astconv/generics.rs

```diff
@@ -459,7 +459,28 @@ impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
 
         let default_counts = gen_params.own_defaults();
         let param_counts = gen_params.own_counts();
-        let named_type_param_count = param_counts.types - has_self as usize;
+        let synth_type_param_count = if tcx.features().explicit_generic_args_with_impl_trait {
```

Comment:
I'd like a small comment here just explaining what we're doing.

Reply:
Done